src/core/server/status/plugins_status.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BehaviorSubject, Observable, combineLatest, of } from 'rxjs';
+import { map, distinctUntilChanged, switchMap, debounce } from 'rxjs/operators';
+import { isDeepStrictEqual } from 'util';
+
+import { PluginName } from '../plugins';
+import { ServiceStatus, CoreStatus } from './types';
+import { getSummaryStatus } from './get_summary_status';
+
+interface Deps {
+  core$: Observable<CoreStatus>;
+  pluginDependencies: ReadonlyMap<PluginName, PluginName[]>;
+}
+
+const waitForMicrotasks = (numberOfTicks: number = 1): Promise<unknown> => {
+  if (numberOfTicks === 1) {
+    return Promise.resolve();
+  } else {
+    return Promise.resolve().then(() => waitForMicrotasks(numberOfTicks - 1));
+  }
+};
+
+export class PluginsStatusService {
+  private readonly pluginStatuses = new Map<PluginName, Observable<ServiceStatus>>();
+  private readonly update$ = new BehaviorSubject(true);
+  constructor(private readonly deps: Deps) {}
+
+  public set(plugin: PluginName, status$: Observable<ServiceStatus>) {
+    this.pluginStatuses.set(plugin, status$);
+    this.update$.next(true); // trigger all existing Observables to update from the new source Observable
+  }
+
+  public getAll$(): Observable<Record<PluginName, ServiceStatus>> {
+    return this.getPluginStatuses$([...this.deps.pluginDependencies.keys()]);
+  }
+
+  public getPlugins$(plugin: PluginName): Observable<Record<PluginName, ServiceStatus>> {
```

Comment:
+1

I think we may also want to consider renaming the `core.status.plugins$` API exposed to plugins (as defined in the RFC). I think it would be more clear if it was `core.status.dependencies$`. WDYT?

Reply:
>  I think it would be more clear if it was core.status.dependencies$

sounds good